code/modules/clothing/shoes/magboots.dm

```diff
@@ -126,3 +126,161 @@
 			..()
 		else
 			to_chat(user, "<span class='notice'>You poke the gem on [src]. Nothing happens.</span>")
+
+
+/obj/item/clothing/shoes/magboots/gravity
+	name = "Gravitational boots"
+	desc = "These experimental boots try to get around the restrictions of magboots by installing miniture gravitational generators in the soles. Sadly, power hungry, and needs a gravitational anomaly core."
+	icon_state = "gravboots0"
+	origin_tech = "materials=6;magnets=6;engineering=6"
+	actions_types = list(/datum/action/item_action/toggle, /datum/action/item_action/ghop) //In other news, combining magboots with jumpboots is a mess
+	strip_delay = 100
+	put_on_delay = 100
+	slowdown_active = SHOES_SLOWDOWN
+	magboot_state = "gravboots"
+	var/datum/martial_art/grav_stomp/style = new //Only works with core and cell installed.
+	var/jumpdistance = 5
+	var/jumpspeed = 3
+	var/recharging_rate = 6 SECONDS
+	var/recharging_time = 0 // Time until next dash
+	var/dash_cost = 1000 // Cost to dash.
+	var/power_consumption_rate = 30 // How much power is used by the boots each cycle when magboots are active
+	var/obj/item/assembly/signaler/anomaly/grav/core = null
+	var/obj/item/stock_parts/cell/cell = null
+
+/obj/item/clothing/shoes/magboots/gravity/Destroy()
+	QDEL_NULL(cell)
+	QDEL_NULL(core)
+	return ..()
+
+/obj/item/clothing/shoes/magboots/gravity/examine(mob/user)
+	. = ..()
+	if(core && cell)
+		. += "<span class='notice'>[src] is fully operational!</span>"
```

Comment:
Otherwise this reads "The gravitational boots is fully operational!"
```suggestion
		. += "<span class='notice'>[src] are fully operational!</span>"
```